packages/at_chops/lib/src/at_chops_impl.dart

```diff
@@ -270,7 +201,8 @@ class AtChopsImpl extends AtChops {
       EncryptionKeyType encryptionKeyType, String? keyName) {
     switch (encryptionKeyType) {
       case EncryptionKeyType.rsa2048:
-        return DefaultEncryptionAlgo(_getEncryptionKeyPair(keyName)!);
+        return DefaultEncryptionAlgo.fromKeyPair(
+            _getEncryptionKeyPair(keyName)!);
       case EncryptionKeyType.rsa4096:
         // TODO: Handle this case.
```

Comment:
Should throw an exception for an unsupported EncryptionKeyType

Reply:
done